### ISSUE TITLE
DefaultPredictor vs build_model - model accuracy discrepancy

### ISSUE THREAD

**Issue by wiamadaya** [facebookresearch/detectron2#941]
i notice accuracy discrepancy when performing inference using `DefaultPredictor `vs `build_model`, am i missing any finer details?

**DefaultPredictor** 
```
cfg = get_cfg()
cfg.merge_from_file("./detectron2/configs/COCO-InstanceSegmentation/mask_rcnn_R_101_FPN_3x.yaml")
cfg.MODEL.ROI_HEADS.SCORE_THRESH_TEST = 0.5
cfg.MODEL.ROI_HEADS.NUM_CLASSES = 6
cfg.MODEL.WEIGHTS = os.path.join(cfg.OUTPUT_DIR, "model_final.pth")

predictor = DefaultPredictor(cfg)
img = cv2.imread("./img (10).jpg")
outputs = predictor(img)
outputs

{'instances': Instances(num_instances=6, image_height=352, image_width=640, fields=[pred_boxes = Boxes(tensor([[243.9448, 282.2408, 276.4154, 349.7356],
         [277.1699, 279.2842, 308.9017, 348.2310],
         [426.8457, 292.2095, 459.3252, 333.8296],
         [312.1853, 276.6370, 344.1364, 345.0952],
         [390.5046, 296.8521, 422.9211, 338.2159],
         [355.2385, 300.6923, 387.5770, 342.5168]], device='cuda:0')), scores = tensor([0.9990, 0.9985, 0.9978, 0.9978, 0.9975, 0.9941], device='cuda:0'), pred_classes = tensor([0, 1, 5, 2, 4, 3], device='cuda:0'), pred_masks = tensor([[[False, False, False,  ...
```

**build_model**
```
cfg = get_cfg()
cfg.merge_from_file("./detectron2/configs/COCO-InstanceSegmentation/mask_rcnn_R_101_FPN_3x.yaml")
cfg.MODEL.ROI_HEADS.SCORE_THRESH_TEST = 0.5
cfg.MODEL.ROI_HEADS.NUM_CLASSES = 6

model = build_model(cfg)
DetectionCheckpointer(model).load('./output/model_final.pth')
model.train(False)
img = cv2.imread("./img (10).jpg")
img = np.transpose(img,(2,0,1))
img_tensor = torch.from_numpy(img)
inputs = [{"image":img_tensor}]
outputs = model(inputs)
outputs

[{'instances': Instances(num_instances=5, image_height=352, image_width=640, fields=[pred_boxes = Boxes(tensor([[357.1407, 300.2568, 387.8150, 345.0016],
          [389.9078, 300.4639, 421.0674, 339.7079],
          [312.0892, 278.7418, 343.8885, 343.2694],
          [244.2871, 281.2459, 275.0525, 350.4953],
          [427.0097, 294.5322, 458.4152, 336.4819]], device='cuda:0',
         grad_fn=<IndexBackward>)), scores = tensor([0.9158, 0.9049, 0.9031, 0.8832, 0.8290], device='cuda:0',
         grad_fn=<IndexBackward>), pred_classes = tensor([3, 4, 2, 0, 5], device='cuda:0'), pred_masks = tensor([[[False, False, False,  ...
```

**Comment by ppwwyyxx** [comment]
There is no claim that they have the same accuracy.

https://detectron2.readthedocs.io/modules/engine.html#detectron2.engine.defaults.DefaultPredictor documents some of the differences.
